detox/src/utils/semanticTypes.test.js

```diff
@@ -0,0 +1,246 @@
+const { getClassNamesForSemanticType, getAvailableSemanticTypes } = require('./semanticTypes');
+
+describe('Enhanced by.type() with semantic types', () => {
+  describe('getClassNamesForSemanticType', () => {
+    it('should return iOS class names for image type', () => {
+      const classNames = getClassNamesForSemanticType('image', 'ios');
+      expect(classNames).toEqual(['RCTImageView', 'UIImageView']);
+    });
+
+    it('should return Android class names for image type', () => {
+      const classNames = getClassNamesForSemanticType('image', 'android');
+      expect(classNames).toEqual(['android.widget.ImageView', 'com.facebook.react.views.image.ReactImageView']);
+    });
+
+    it('should return iOS class names for input-field type', () => {
+      const classNames = getClassNamesForSemanticType('input-field', 'ios');
+      expect(classNames).toEqual(['RCTTextInput', 'RCTMultilineTextInput', 'UITextField', 'UITextView']);
+    });
+
+    it('should return Android class names for input-field type', () => {
+      const classNames = getClassNamesForSemanticType('input-field', 'android');
+      expect(classNames).toEqual(['android.widget.EditText', 'com.facebook.react.views.textinput.ReactEditText']);
+    });
+
+    it('should throw error for unknown semantic type', () => {
+      expect(() => {
+        getClassNamesForSemanticType('unknown-type', 'ios');
+      }).toThrow('Unknown semantic type: unknown-type');
+    });
+
+    it('should throw error for unsupported platform', () => {
+      expect(() => {
+        getClassNamesForSemanticType('image', 'windows');
+      }).toThrow('Platform windows not supported for semantic type image');
+    });
+
+    it('should return iOS class names for scrollview type', () => {
+      const classNames = getClassNamesForSemanticType('scrollview', 'ios');
+      expect(classNames).toEqual(['RCTScrollView', 'UIScrollView']);
+    });
+
+    it('should return Android class names for scrollview type', () => {
```

Comment:
I think we could implement a for-each over a params matrix (type, platform) and perform a Jest snapshot-match assertion, and that would do it.